packages/builder/src/components/backend/DatasourceNavigator/modals/DatasourceConfigModal.svelte

```diff
@@ -15,11 +22,25 @@
   // kill the reference so the input isn't saved
   let datasource = cloneDeep(integration)
   let isValid = false
+  let fetchTableStep = false
+  let selectedTables = []
+  let tableList = []
 
+  $: console.log(selectedTables)
   $: name =
-    IntegrationNames[datasource.type] || datasource.name || datasource.type
+    IntegrationNames[datasource?.type] || datasource?.name || datasource?.type
+  $: datasourcePlus = datasource?.plus
+  $: title = fetchTableStep ? "Fetch your tables" : `Connect to ${name}`
+  $: confirmText = fetchTableStep
+    ? "Continue"
+    : datasourcePlus
+    ? "Connect"
+    : "Save and continue to query"
 
   async function validateConfig() {
+    if (!integration.features[DatasourceFeature.CONNECTION_CHECKING]) {
```

Comment:
integration.features undefined will break the js (as the live bug yesterday)

Reply:
ah yes - good catch!